src/Native/NativeUi.js

```diff
@@ -387,6 +387,49 @@ var _ohanhi$elm_native_ui$Native_NativeUi = (function () {
     };
   }
 
+  /**
+   * Run the provided `flagDecoder` before passing to Elm code for type safety.
+   */
+  function programWithFlags(impl) {
+    return function(flagDecoder) {
+      return function(object, moduleName, debugMetadata) {
+        object.start = function start(onAppReady, flags = {}) {
+          if (typeof flagDecoder === 'undefined') {
+            throw new Error(
+              'Are you trying to sneak a Never value into Elm? Trickster!\n'
+              + 'It looks like ' + moduleName + '.main is defined with `programWithFlags` but has type `Program Never`.\n'
+              + 'Use `program` instead if you do not want flags.'
+            );
+          }
+          var result = A2(_elm_lang$core$Native_Json.run, flagDecoder, flags);
+          if (result.ctor === 'Err')
+          {
+          	throw new Error(
+          		moduleName + '.worker(...) was called with an unexpected argument.\n'
+          		+ 'I tried to convert it to an Elm value, but ran into this problem:\n\n'
+          		+ result._0
+          	);
+          }
+          return makeComponent(impl, onAppReady, result);
+        };
+      };
+    };
+  }
+
+  /**
+   * This is unsafe since it passes `flags` object wihout running the `flagDecoder`.
+   * Useful for passing down injected JS props to another JS functions in native codes.
+   */
+  function unsafeProgramWithFlags(impl) {
+    return function(flagDecoder) {
```

Comment:
HI, Thank you for the acceptance. 

No it is not, it exists just for the consistency. The `program` function in the [elm-lang/core](https://github.com/elm-lang/core/blob/master/src/Native/Platform.js#L10) also has it.